src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -10601,10 +10603,16 @@ and TcAndPatternCompileMatchClauses mExpr mMatch actionOnFailure cenv inputExprO
     let matchVal, expr = CompilePatternForMatchClauses cenv env mExpr mMatch true actionOnFailure inputExprOpt inputTy resultTy.Commit clauses
     matchVal, expr, tpenv
 
-and TcMatchPattern cenv inputTy env tpenv (synPat: SynPat) (synWhenExprOpt: SynExpr option) =
+and TcMatchPattern (cenv: cenv) inputTy env tpenv (synPat: SynPat) (synWhenExprOpt: SynExpr option) (isTrueMatchClause: bool) =
```

Comment:
Instead of introducing booleans, we should probably use a struct DU, as it's much easier to read when you looking at such code (like `Yes | No` boolean in cases of "boolean-like" values).

Reply:
Done